docs/src/main/asciidoc/cdi-integration.adoc

```diff
@@ -0,0 +1,544 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/master/docs/src/main/asciidoc
+////
+= Quarkus - CDI Integration Guide
+
+include::./attributes.adoc[]
+:numbered:
+:sectnums:
+:sectnumlevels: 4
+:toc:
+:toclevels: 2
+
+The ArC container is bootstrapped at build time.
+The downside of this approach is that CDI Portable Extensions cannot be supported.
+Nevertheless, the functionality can be achieved using the Quarkus-specific extensions API.
+
+The container is bootstrapped in multiple phases.
+From a high level perspective these phases go as follows:
+
+1. Initialization
+2. Bean discovery 
+3. Registration of synthetic components
+4. Validation
+
+In the _initialization_ phase the preparatory work is being carried out and custom contexts are registered.
+_Bean discovery_ is then the process where the container analyzes all application classes, identifies beans and wires them all together based on the provided metadata.
+Subsequently, the extensions can register _synthetic components_.
+Attributes of these components are fully contolled by the extensions, i.e. are not derived from an existing class.
```

Comment:
```suggestion
Attributes of these components are fully controlled by the extensions, i.e. are not derived from an existing class.
```

It would also be helpful to add an example of what is meant by "not derived from an existing class" to make synthetic beans more palatable to users. In particular, when they make sense as opposed to "regular" beans…